### ISSUE TITLE
www-apps/gitlab-ce-8.5.0: bundler failed

### ISSUE THREAD

**Issue by Kami-no** [fritteli/gentoo-overlay#33]
Installing haml 4.0.7
Installing fog-core 1.35.0
An error occurred while installing rugged (0.24.0b13), and Bundler cannot continue.
Make sure that `gem install rugged -v '0.24.0b13'` succeeds before bundling.
 * ERROR: www-apps/gitlab-ce-8.5.0::fritteli failed (install phase):
 *   bundler failed
 * 
 * Call stack:
 *     ebuild.sh, line  133:  Called src_install
 *   environment, line 4680:  Called ruby-ng_src_install
 *   environment, line 4484:  Called _ruby_invoke_environment 'all' 'all_ruby_install'
 *   environment, line  540:  Called all_ruby_install
 *   environment, line  640:  Called die
 * The specific snippet of code:
 *       ${RUBY} /usr/bin/bundle install ${bundle_args} || die "bundler failed";
 * 
 * If you need support, post the output of `emerge --info '=www-apps/gitlab-ce-8.5.0::fritteli'`,
 * the complete build log and the output of `emerge -pqv '=www-apps/gitlab-ce-8.5.0::fritteli'`.
!!! When you file a bug report, please include the following information:
GENTOO_VM=  CLASSPATH="" JAVA_HOME=""
JAVACFLAGS="" COMPILER=""
and of course, the output of emerge --info =gitlab-ce-8.5.0
 * The complete build log is located at '/var/tmp/portage/www-apps/gitlab-ce-8.5.0/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/www-apps/gitlab-ce-8.5.0/temp/environment'.
 * Working directory: '/var/tmp/portage/www-apps/gitlab-ce-8.5.0/image/opt/gitlab'
 * S: '/var/tmp/portage/www-apps/gitlab-ce-8.5.0/work/all/gitlabhq-8.5.0'

**Comment by tkurbad** [comment]
The "real" error occurs much earlier in the build:

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    /usr/bin/ruby21 extconf.rb
checking for gmake... yes
Building Rugged using system libraries.
libgit2 version is not compatible, expected ~> 0.23.0
```

In fact, the build doesn't accept the installed libgit2-0.24.0_rc1 as valid, nor would it compile with the older 0.23.x versions.

The easiest way to resolve the issue seems not to set the 'rugged_use_system_libraries' USE flag on the gitlab-ce ebuild, although this counterfeits Gentoo philosophy not to allow private library versions.

Best,
Torsten

**Comment by fritteli** [comment]
Hi all
I'm sorry, I seem to have created a faulty ebuild for libgit2.
@tkurbad is right, try disabling the rugged_use_system_libraries USE flag as a quick-fix. Meanwhile, I'm working on new ebuilds that should fix the build failure.

Sorry for the inconvenience.

**Comment by fritteli** [comment]
I've just pushed new ebuilds of `gitlab-ce`and `libgit2`.
On my server (https://gittr.ch), I've successfully installed `gitlab-ce-8.5.0-r1` with `USE=-rugged_use_system_libraries`, so I don't know if the ebuilds for `libgit2` actually works. If in doubt, disable the rugged-USE-flag.

I'll close this issue, as I could successfully upgrade. If you still encounter problems, either open a new issue or re-open this one.

And thanks a lot for reporting the error!
